presto-google-sheets/src/main/java/io/prestosql/plugin/google/sheets/SheetsClient.java

```diff
@@ -184,12 +227,28 @@ public SheetsClient(SheetsConfig config, JsonCodec<Map<String, List<SheetsTable>
             if (data.get(i).size() >= 2) {
                 String tableId = String.valueOf(data.get(i).get(0));
                 String sheetId = String.valueOf(data.get(i).get(1));
+                if (data.get(i).size() >= 3 && isArchived(String.valueOf(data.get(i).get(3)))) {
```

Comment:
if date.get(i).size() == 3 exactly, then data.get(i).get(3) would actually fail right? the max index is 2 in this case. similarly for the >= 4 below